AIO/src/org/aio/activities/quests/SheepShearer.java

```diff
@@ -0,0 +1,150 @@
+package org.aio.activities.quests;
+
+import org.aio.activities.activity.Activity;
+import org.aio.activities.banking.DepositAllBanking;
+import org.aio.util.MakeAllInterface;
+import org.aio.util.Sleep;
+import org.osbot.rs07.api.map.Area;
+import org.osbot.rs07.api.model.GroundItem;
+import org.osbot.rs07.api.model.NPC;
+import org.osbot.rs07.api.model.RS2Object;
+import org.osbot.rs07.api.ui.Tab;
+
+public class SheepShearer extends QuestActivity {
+
+    private MakeAllInterface makeAllInterface;
+
+    private static final Area FARMER = new Area(3188, 3275, 3190, 3270);
+    private static final Area SHEEP = new Area(3195, 3272, 3209, 3260);
+    private static final Area SPINNER = new Area(3209, 3215, 3211, 3212).setPlane(1);
+
+    private static final String[] FARMER_OPTIONS = {
+            "I'm looking for a quest.",
+            "Yes okay. I can do that.",
+            "Of course!",
+            "I'm something of an expert actually!"
+    };
+
+    private static final int INVENTORY_SLOTS_REQUIRED = 22;
+
+    private static final String[] ITEMS_NEEDED = {
+            "Ball of wool",
+            "Wool",
+            "Shears"
+    };
+
+    private final DepositAllBanking depositAllBanking = new DepositAllBanking(ITEMS_NEEDED);
+
+    public SheepShearer() {
+        super(Quest.SHHEP_SHEARER);
+    }
+
+    @Override
+    public void onStart() {
+        depositAllBanking.exchangeContext(getBot());
+
+        makeAllInterface = new MakeAllInterface("Ball of Wool");
+        makeAllInterface.exchangeContext(getBot());
+    }
+
+    @Override
+    public void runActivity() throws InterruptedException {
+        if (getInventory().getEmptySlotCount() < INVENTORY_SLOTS_REQUIRED - getInventory().getAmount(ITEMS_NEEDED)) {
+            depositAllBanking.run();
+        } else if (getTabs().getOpen() != Tab.INVENTORY) {
+            getTabs().open(Tab.INVENTORY);
+        } else {
+            switch (getProgress()) {
+                case 0:
+                    talkToFarmer();
+                    break;
+                case 1:
+                    if (hasRequiredItems()) {
+                        talkToFarmer();
+                    } else {
+                        getItemsNeeded();
+                    }
+                    break;
+                case 21:
+                    log("Quest is complete");
```

Comment:
Good work. In addition to logging, how do you feel about using the setStatus() function? It renders more information in the actual client about what is happening.

Reply:
I don't know much about the `setStatus()` function, I will look into it over the weekend, I would also like to have the activity show what quest is being done, and maybe have it at an even lower level so it shows what its doing in the quest currently based on the `getProgress()`